pkg/identityserver/gormstore/gateway_store.go

```diff
@@ -213,6 +213,20 @@ func (s *gatewayStore) DeleteGateway(ctx context.Context, id *ttnpb.GatewayIdent
 
 func (s *gatewayStore) RestoreGateway(ctx context.Context, id *ttnpb.GatewayIdentifiers) error {
 	defer trace.StartRegion(ctx, "restore gateway").End()
+	if len(id.Eui) > 0 {
+		query := s.query(ctx, Gateway{}, withGatewayEUI(EUI64(types.MustEUI64(id.Eui).OrZero())))
+		query = selectGatewayFields(ctx, query, []string{"ids.gateway_id", "ids.eui"})
+		var gtwModel Gateway
+		if err := query.First(&gtwModel).Error; err != nil {
+			if gorm.IsRecordNotFoundError(err) {
+				return s.restoreEntity(ctx, id)
+			}
+			return err
+		}
+		if gtwModel.GatewayID != id.GatewayId {
+			ctx = context.WithValue(ctx, cleanGtwEuiKey, true)
```

Comment:
You can already update the EUI to `nil` here, right ? No need to do the context enrichment and do this in `restoreEntity`.

Reply:
Removed it